purchases/src/test/java/com/revenuecat/purchases/google/usecase/ConsumePurchaseUseCaseTest.kt

```diff
@@ -0,0 +1,1244 @@
+            ConsumePurchaseUseCaseParams(
+                "purchaseToken",
+                PostReceiptInitiationSource.PURCHASE,
+                appInBackground = false,
+            ),
+            { _ ->
+                Assertions.fail("shouldn't be success")
+            },
+            { error ->
+                receivedError = error
+            },
+            withConnectedClient = {
+                timesRetried++
+                it.invoke(mockClient)
+            },
+            executeRequestOnUIThread = { _, request ->
+                consumeStubbing answers {
+                    slot.captured.onConsumeResponse(
+                        BillingClient.BillingResponseCode.SERVICE_UNAVAILABLE.buildResult(),
+                        "purchaseToken"
+                    )
+                }
+
+                request(null)
+            },
+        )
+
+        useCase.run()
+
+        assertThat(timesRetried).isEqualTo(1)
+        assertThat(receivedError).isNotNull
+        assertThat(receivedError!!.code).isEqualTo(PurchasesErrorCode.StoreProblemError)
+    }
+
+    @Test
+    fun `If service returns SERVICE_UNAVAILABLE, re-execute with backoff for unsynced active purchases`() {
+        val slot = slot<ConsumeResponseListener>()
+        val consumeStubbing = every {
+            mockClient.consumeAsync(
+                any(),
+                capture(slot),
+            )
+        }
+        var receivedError: PurchasesError? = null
+        val capturedDelays = mutableListOf<Long>()
+        val useCase = ConsumePurchaseUseCase(
+            ConsumePurchaseUseCaseParams(
+                "purchaseToken",
+                PostReceiptInitiationSource.UNSYNCED_ACTIVE_PURCHASES,
+                appInBackground = true,
+            ),
+            { _ ->
+                Assertions.fail("shouldn't be success")
+            },
+            { error ->
+                receivedError = error
+            },
+            withConnectedClient = {
+                it.invoke(mockClient)
+            },
+            executeRequestOnUIThread = { delay, request ->
+                capturedDelays.add(delay)
+                consumeStubbing answers {
+                    slot.captured.onConsumeResponse(
+                        BillingClient.BillingResponseCode.SERVICE_UNAVAILABLE.buildResult(),
+                        "purchaseToken"
+                    )
+                }
+
+                request(null)
+            },
+        )
+
+        useCase.run()
+
+        assertThat(capturedDelays.size).isEqualTo(12)
+        assertThat(capturedDelays.last()).isCloseTo(RETRY_TIMER_MAX_TIME_MILLISECONDS, Offset.offset(1000L))
+        assertThat(receivedError).isNotNull
+        assertThat(receivedError!!.code).isEqualTo(PurchasesErrorCode.StoreProblemError)
+    }
+
+    @Test
+    fun `If service returns SERVICE_UNAVAILABLE, don't retry and error if user in session for unsynced active purchases`() {
```

Comment:
Some of these tests are kinda duplicated among the use case tests... I guess we could try with paremeterized tests but not sure if it's worth trying to extract them for now.